### PR TITLE
Added CsvInvalidEvents metric to CSV Processor

### PR DESCRIPTION
**Pull request by finnroblin**
Signed-off-by: Finn Roblin <finnrobl@amazon.com>

### Description
Quick change to add a metric that tracks the number of invalid events passed through the CSV Processor. An invalid event is any event that causes the Jackson csv parser to throw an exception. The parser will only throw exceptions if there's an unclosed quote or other invalid character. The processor now increments the counter and continues to the next event. 
 
### Issues Resolved
Not tracked in any issues.
 
### Check List
- [X] New functionality includes testing.
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by travisbenedict — APPROVED** [review]
LGTM

File: data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CsvProcessor.java [hunk 1]
```diff
@@ -67,6 +74,7 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     event.delete(config.getColumnNamesSourceKey());
                 }
             } catch (final IOException e) {
+                csvInvalidEventsCounter.increment();
```
Comment (asifsmohammed) [inline data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CsvProcessor.java:77]: I'm not sure if I understood this, how is IOException related to `csvInvalidEventsCounter`?

Comment (finnroblin) [reply data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CsvProcessor.java:77]: Jackson's `readValues`, `hasNextValue`, and `nextValue` methods throw a generic `IOException` when there's some issue with the input they're parsing. In testing, I could only get an IOException to throw if the csv record was malformed in some way such as forgetting to close a quote. So I figured that an IOException is a catch-all for invalid CSV Events. Do you think I should catch a more specific error? Perhaps a JSON Parse exception might make this function more readable.

Comment (asifsmohammed) [reply data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CsvProcessor.java:77]: I think it's fine if Jackson throws IOException.

**Review by asifsmohammed — APPROVED** [review]

**Pull request merged**